### ISSUE TITLE
Does flatend support broadcasting a message to all clients in a room, like chat?

### ISSUE THREAD

**Issue by didip** [lithdew/flatend#7]
This framework looks very promising with a very nice interface.

I am curious if I can use it for chat like application.

**Comment by lithdew** [comment]
With the current codebase, there is only support for sending a single message to a single node serving a specific service (we're still figuring out how to expose the message sending API properly so that it is flexible enough for any kind of broadcast pattern like the one you described).

That being said, I can expose a method in the Go SDK right now to let you grab all available nodes for a single service that a node knows of, so that you can easily broadcast a message to all nodes providing a specific service. This way, you can easily use Flatend for something like a chat app (broadcast to all nodes in a chatroom).

What do you think?

The code would look something like:

```go
nodes := node.ProvidersFor("chatroom")

for _, node := range nodes {
    _, _ := node.Push([]string{"chatroom"}, nil, strings.NewReader("chat message here!"))
}
```

**Comment by didip** [comment]
Wow, super quick response!

The proposal looks great! I can work with it.

**Comment by lithdew** [comment]
![image](https://user-images.githubusercontent.com/63115601/85885407-3facaf00-b81f-11ea-8390-3e761df2420c.png)

Almost ready, will push soon :smile:.

**Comment by lithdew** [comment]
Released v0.0.8 which has the `ProvidersFor` method available in the Go and NodeJS SDK: https://github.com/lithdew/flatend/releases/tag/v0.0.8

The example for broadcasting to all providers is written up and available here: https://github.com/lithdew/flatend/tree/master/examples/go/broadcast

Enjoy :smile:.

**Comment by didip** [comment]
The velocity is off the chart! Thank you for considering this feature!
